src/mongocxx/collection.cpp

```diff
@@ -207,7 +207,7 @@ cursor collection::find(view_or_value filter, const options::find& options) {
     }
 
     if (options.max_time()) {
-        filter_builder << "$maxTimeMS" << options.max_time()->count();
+        filter_builder << "$maxTimeMS" << bsoncxx::types::b_int64{options.max_time()->count()};
```

Comment:
I think it would make more sense to put the runtime check in the option class, so the exception triggers when set.